qemu/tests/qemu_iotests.py

```
@@ -1,50 +1,86 @@
+    if os.path.exists(destination_dir):
+        shutil.rmtree(destination_dir)
+    os.makedirs(destination_dir)
+    try:
+        repo = git.GitRepoHelper(uri=uri, branch=branch, lbranch=lbranch,
+                                 commit=commit,
+                                 destination_dir=destination_dir,
+                                 base_uri=base_uri)
+        repo.git_cmd("clone %s %s" % (uri, destination_dir))
+        repo.git_cmd("checkout %s" % branch)
+        repo.git_cmd("submodule update --init %s" % submodule)
+        logging.info("Generate common.env")
+        os.chdir(destination_dir)
+        cmd = "./configure"
+        if image_format == "raw":
+            cmd += " && make tests/qemu-iotests/socket_scm_helper"
+        process.system(cmd, shell=True)
 
-    error.context("running qemu-iotests for image format %s" % image_format)
-    utils.system("%s -%s" % (cmd, image_format))
+        error_context.context("running qemu-iotests for image format %s"
+                              % image_format, logging.info)
+        os.environ["QEMU_PROG"] = utils_misc.get_qemu_binary(params)
+        os.environ["QEMU_IMG_PROG"] = utils_misc.get_qemu_img_binary(params)
+        os.environ["QEMU_IO_PROG"] = utils_misc.get_qemu_io_binary(params)
+        os.chdir(os.path.join(destination_dir, iotests_dir))
+        cmd = './check'
+        if extra_options:
+            cmd += " %s" % extra_options
+        cmd += " -%s" % image_format
+        output = process.system_output(cmd, ignore_status=True, shell=True)
+        match = re.search(result_pattern, output, re.I | re.M)
+        if match:
+            iotests_log_file = "qemu_iotests_%s.log" % image_format
+            iotests_log_file = utils_misc.get_path(test.debugdir, iotests_log_file)
+            file(iotests_log_file, "w+").write(output)
+            msg = "Total test %s cases, %s failed"
+            raise exceptions.TestFail(msg % (match.group(2), match.group(1)))
+    finally:
```

need ignore errors in finally, or it will cover the real errors